addons/l10n_ar_withholding/wizards/account_payment_register.py

```diff
@@ -89,3 +93,29 @@ def _get_conversion_rate(self):
                 self.payment_date,
             )
         return 1.0
+
+    @api.depends('partner_id', 'payment_date')
+    def _compute_l10n_ar_withholding_ids(self):
+        """ Compute (AR) withholding on payments. """
+        date = fields.Date.from_string(self.payment_date) or datetime.date.today()
+        is_supplier_payment = False
+        active_model = self._context['active_model']
+        if active_model == 'account.move.line':
+            is_supplier_payment = self.line_ids.move_id.is_purchase_document()
+        partner_taxes = self.env['l10n_ar.partner.tax'].search([
+            '|', ('from_date', '>=', date), ('from_date', '=', False),
+            '|', ('to_date', '<=', date), ('to_date', '=', False),
+            ('partner_id', '=', self.partner_id.commercial_partner_id.id),
+            ('company_id', '=', self.company_id.id),
+            ('tax_id.l10n_ar_withholding_payment_type', '=', 'supplier' if is_supplier_payment else 'customer')
+        ])
```

Comment:
I think this field should only be on the parent and not sure if it would work with the _commercial_fields

Reply:
Maybe only when is_company, so then we need child_of and check_company_domain here as well.  (I wonder if cache wise it is not better to do self.partner_id.commercial_partner_id.l10n_ar_partner_tax_ids.filtered(

Reply:
@pablohmontenegro Let's go back to this:
`('partner_id', '=', self.partner_id.commercial_partner_id.id),`